src/packets.rs

```diff
@@ -202,7 +200,7 @@ impl<'a> From<ReasonCode> for Reason<'a> {
         Self {
             reason: Some(ReasonData {
                 code,
-                _properties: None,
+                _properties: Some(Properties::Slice(&[])),
```

Comment:
Out of curiosity: Why that change? ~Can't we serialize `None`?~

Reply:
There's some strangeness in `None` for the serializer because MQTT allows for things to be entirely omitted. The serializer treats `None` as _literally nothing to serialize_. So no data gets serialized at all. However, the MQTT spec requires that we serialize the varint of the properties length (a single zero). So in this case, `None` serializes to `[]`, but `Some(Properties::Slice(&[]))` serializes to `[0]`. It's funky, but flexible.